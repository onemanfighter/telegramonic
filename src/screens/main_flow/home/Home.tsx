import { SearchStack } from '@components';
import { HomeHeaderComponent } from './component';
import { Box } from '@chakra-ui/react';

const Home = () => {
  const onSearchClick = (searchText: string) => {
    console.log(searchText);
  };

  return (
    <Box width={'100%'}>
      <HomeHeaderComponent />
      <SearchStack
        onSearchClick={() => {
          onSearchClick('searchText');
        }}
      />
    </Box>
  );
};

export default Home;
