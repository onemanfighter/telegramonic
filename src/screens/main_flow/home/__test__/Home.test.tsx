import { render } from '@testing-library/react';
import Home from '../Home';

describe('Home screen', () => {
  it('should render correctly', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
