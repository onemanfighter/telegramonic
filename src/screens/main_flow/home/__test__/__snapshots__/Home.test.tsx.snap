// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen should render correctly 1`] = `
<div>
  <div
    class="css-8atqhb"
  >
    <div
      class="css-8atqhb"
    >
      <div
        class="css-77af5c"
      >
        <p
          class="chakra-text css-x3elb1"
        >
          Home.header.title
        </p>
        <p
          class="chakra-text css-1uf6a8x"
        >
          Home.header.subTitle
        </p>
      </div>
    </div>
    <div
      class="css-xkotu"
    >
      <div
        class="chakra-stack css-p09dns"
      >
        <div
          class="chakra-input__group css-15tlywb"
          data-group="true"
        >
          <div
            class="chakra-input__left-element css-1u2k0bl"
          >
            <svg
              color="gray.300"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <line
                x1="21"
                x2="16.65"
                y1="21"
                y2="16.65"
              />
            </svg>
          </div>
          <input
            class="chakra-input css-0"
            placeholder="Search.searchInputPlaceholder"
            type="text"
          />
        </div>
        <button
          aria-controls="menu-list-:r1:"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-button chakra-menu__menu-button css-f24fvb"
          id="menu-button-:r1:"
          type="button"
        >
          <span
            class="css-xl71ch"
          >
            Search.searchCategory.placeholder
          </span>
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              />
            </svg>
          </span>
        </button>
        <div
          class="css-1ow4kgd"
          style="visibility: hidden; position: absolute; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-12m5n0v"
            id="menu-list-:r1:"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="0"
              id="menu-list-:r1:-menuitem-:r2:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Search.searchCategory.options.all
            </button>
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="1"
              id="menu-list-:r1:-menuitem-:r3:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Search.searchCategory.options.bots
            </button>
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="2"
              id="menu-list-:r1:-menuitem-:r4:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Search.searchCategory.options.channels
            </button>
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="3"
              id="menu-list-:r1:-menuitem-:r5:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Search.searchCategory.options.groups
            </button>
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="4"
              id="menu-list-:r1:-menuitem-:r6:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Search.searchCategory.options.stickers
            </button>
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="5"
              id="menu-list-:r1:-menuitem-:r7:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Search.searchCategory.options.themes
            </button>
          </div>
        </div>
        <button
          class="chakra-button css-un5p2g"
          type="button"
        >
          Search.searchButton
        </button>
      </div>
    </div>
  </div>
</div>
`;
